src/main/java/net/md_5/specialsource/repo/JarRepo.java

```diff
@@ -48,7 +48,12 @@ protected ClassNode findClass0(String internalName) {
             InputStream is = jar.getClass(internalName);
 
             if (is != null) {
-                ClassReader reader = new ClassReader(is);
+                ClassReader reader;
+                try {
+                    reader = new ClassReader(is);
+                } finally {
+                    is.close();
```

Comment:
Double closing the stream now, .close is called again a few lines below. Really this entire method should just be a call to jar.getNode, the code is duplicated currently.

Reply:
My bad. Will refactor it as suggested.